### DISCUSSION TITLE
Can't swap transactions on pancake.

### DISCUSSION THREAD

**Discussion by 2jumper3** [WalletConnect/walletconnect-monorepo#4065] [Developer Support] [answered]
Hi all! 
I'm complete the integration on https://react-app.walletconnect.com and trying to check it on production. 
Right now I'm connect my wallet to [pancakeswap](https://pancakeswap.finance), and can't understand why it doesn't work. 
I'm trying to do it with TrustWallet too, and that's what I got: 
https://bscscan.com/tx/0x9e003dc20d8bf0ac65575b902ba3f6d838aa8df4f240daa0c3d402337d4073c3
Looks good. And work great. 
And my transaction with my implementation: 
https://bscscan.com/tx/0x34d0bc404dea931d56fbd41cae875021e0a37027fa6dffb77f0eb55df3fa837f

The method which pancake send to me: "eth_sendTransaction". So I send it to blockchain and response to walletConnect with hash. But dApp return to me error and this failed transaction... 
Can't understand what happened... 

Plus I've receive huge `data` in wallet connect sessionRequestPublisher, and don't understand what I should do with it ) in https://react-app.walletconnect.com this filed is empty... 
<img width="1391" alt="Screenshot 2023-12-20 at 12 10 12" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/45697468/640716c4-d911-4e20-be76-f55fbf1293b5">

**Comment by glitch-txs** [comment, marked as answer]
you need to promp the user to sign the transaction and send the signed tx to the blockchain.
